commit/tokenprice/observation.go

```diff
@@ -0,0 +1,110 @@
+package tokenprice
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/smartcontractkit/chainlink-ccip/shared"
+
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	"golang.org/x/exp/maps"
+)
+
+func (p *processor) ObserveFDestChain() (int, error) {
+	fChain, err := p.homeChain.GetFChain()
+	if err != nil {
+		// TODO: metrics
+		p.lggr.Warnw("call to GetFChain failed", "err", err)
+		return 0, fmt.Errorf("failed to get FChain")
+	}
+
+	fDestChain, ok := fChain[p.cfg.DestChain]
+	if !ok {
+		return 0, fmt.Errorf("fChain does not have dest chain")
+	}
+
+	return fDestChain, nil
+}
+
+func (p *processor) ObserveFeedTokenPrices(ctx context.Context) []cciptypes.TokenPrice {
+	supportedChains, err := p.chainSupport.SupportedChains(p.oracleID)
```

Comment:
Food for thought, I see this pattern a lot across other processors, shouldn't we extract that then, let's say to something like this and call that from the upper layer (probably plugin) instead of doing these checks in every underlying processor? 

```go
type Processor interface {
   IsChainSupported() (bool, error)
}
```

```go
func (p *plugin) Observer() {
    if !p.TokenProcessor.IsChainSupported() {
       return sth
    }
    if !p.MerkleRootProcessor.IsChainSupported() {
       return sth
    }

}```

Reply:
Going to postpone this to another PR for now.